internal/cosesign1/cosesign1util_test.go

```diff
@@ -0,0 +1,169 @@
+//go:build linux
+// +build linux
+
+package cosesign1
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/veraison/go-cose"
+)
+
+func readFileBytes(filename string) ([]byte, error) {
+	content, err := os.ReadFile(filename)
+	if err != nil {
+		println("Error reading '" + filename + "': " + string(err.Error()))
+		return nil, err
+	}
+	if len(content) == 0 {
+		println("Warning: empty file '" + filename + "'")
+	}
+	return content, nil
+}
+
+func readFileString(filename string) (string, error) {
+	data, err := readFileBytes(filename)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+var fragmentRego string
+var fragmentCose []byte
+var leafPrivatePem string
+var leafCertPEM string
+var leafPubkeyPEM string
+var certChainPEM string
+
+func comparePEMs(pk1pem string, pk2pem string) bool {
+	pk1der := pem2der([]byte(pk1pem))
+	pk2der := pem2der([]byte(pk2pem))
+	return bytes.Equal(pk1der, pk2der)
+}
+
+// Decode a COSE_Sign1 document and check that we get the expected payload, issuer, keys, certs etc.
+func Test_UnpackAndValidateCannedFragment(t *testing.T) {
+	var unpacked *UnpackedCoseSign1
+	unpacked, err := UnpackAndValidateCOSE1CertChain(fragmentCose, nil, nil, false)
+
+	if err != nil {
+		t.Errorf("UnpackAndValidateCOSE1CertChain failed: %s", err.Error())
+	}
+
+	iss := unpacked.Issuer
+	feed := unpacked.Feed
+	pubkey := base64PublicKeyToPEM(unpacked.Pubkey)
+	pubcert := base64CertToPEM(unpacked.Pubcert)
+	payload := string(unpacked.Payload[:])
+	cty := unpacked.ContentType
+
+	if !comparePEMs(pubkey, leafPubkeyPEM) {
+		t.Error("pubkey did not match")
+	}
+	if !comparePEMs(pubcert, leafCertPEM) {
+		t.Error("pubcert did not match")
+	}
+	if cty != "application/unknown+json" {
```

Comment:
Since you use predefined variables for the pubkey and pubcert comparison, why not do the same here instead of hard-coded strings?

Reply:
The content type is fixed. The certs will be different across runs. Even if we move back to fixed certs across runs (getting past the credential checks done by source control systems, then the certs are still arbitrary and made up.